### PR TITLE
Github support: TEXT mode and UTF-8 support for all objects

### PR DESCRIPTION
**Pull request by gilbertbw**
As discussed in #25, add support to work with a plain folder, rather than SVN. This allows the use of any SCC system (including git). This is refered to as TEXT mode (in contrast to SVN mode).

In addition there is now support for encoding all files as UTF-8, rather than just UCS2 encoded files. This helps when working with SCC tools that expect UTF-8 (rather than CP1252).

These changes make it possible to the the add-in to develop projects hosted on Github.

[AccessSCC.zip](https://github.com/x-ware-ltd/access-scc-addin/files/5509422/AccessSCC.zip)

### CODE REVIEW

**Comment by OliverVahey** [comment]
Newer version of addin
[AccessSCC.zip](https://github.com/x-ware-ltd/access-scc-addin/files/5511731/AccessSCC.zip)

File: Forms/frmSplash.ACF [hunk 1]
```diff
@@ -112,1121 +112,26 @@ Begin Form
                 0xe12fa329ea7bc540ae94cf4ebf07048a
             End
             Begin
-                Begin Image
```
Comment (bkp7) [inline Forms/frmSplash.ACF:115]: Please can you put an image of some sort back in. It looks awful without it. Perhaps https://favpng.com/png_view/fork-branching-source-code-version-control-fork-png/mce0WrLp

File: Forms/Access SVN - Options.ACF [hunk 2]
```diff
@@ -1088,1121 +1132,10 @@ Begin Form
                     WebImagePaddingRight =1
                     WebImagePaddingBottom =1
                 End
-                Begin Label
```
Comment (bkp7) [inline Forms/Access SVN - Options.ACF:1091]: Please put an image back - this? https://favpng.com/png_view/fork-branching-source-code-version-control-fork-png/mce0WrLp

**Review by bkp7 — CHANGES_REQUESTED** [review]
USysRegInfo Table needs to be updated.
Also after creating the new mda you need to populate the 'Company' and 'Comments' fields in the DB Properties section (File/'View and edit database properties'). The values for this need to be decided

File: README.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ Unlike the Visual Sourcesafe Addin
 
 *Stop the press*: As of Access 2013, the SCC layer of Access has been "deprecated". This means if you are already using systems that rely on the MSSCCI layer, such as Visual Sourcesafe, [PushOK SCC plugin](http://www.pushok.com/software/svn.html) or [Unified SCC](http://aigenta.com/products/UnifiedScc.aspx), this will no longer work from Access 2013 onwards.
 
-### Recent Release History
+# Recent Release History
```
[the remark this answers is not in the document]
Comment (gilbertbw) [reply README.md:14]: replaced with link to releases page

File: README.md [hunk 4]
```diff
@@ -24,10 +24,10 @@ Unlike the Visual Sourcesafe Addin
 0.47 
 - Update to handle regional formatting of numbers. Prior to this release some regions may have caused errors, and would have stored numeric table data incorrectly in the repository. Thanks to Florian Reitmeit for spotting this one.
 
-### Version Control Systems
+# Version Control Systems
```
[the remark this answers is not in the document]
Comment (gilbertbw) [reply README.md:27]: Done

File: README.md [hunk 5]
```diff
@@ -24,10 +24,10 @@ Unlike the Visual Sourcesafe Addin
...
 The only VCS that is supported at the moment is Subversion but future plans could allow support for Git, Mercurial, possibly TFS (Team Foundation Server) and many others if there is enough demand.
 
-### Access Versions
+# Access Versions
```
[the remark this answers is not in the document]
Comment (gilbertbw) [reply README.md:30]: I think this is correct. We still build a 2002-2003 format database.

File: Modules/modSCC.ACM [hunk 6]
```diff
@@ -5,6 +5,7 @@ Public Enum SccSystem
   sccSystemSubversion = 1  ' Supported
   sccSystemGIT = 2         ' Unsupported
   sccSystemMercurial = 3   ' Unsupported
+  sccSystemText = 4        ' Supported
```
[the remark this answers is not in the document]
Comment (gilbertbw) [reply Modules/modSCC.ACM]: Done

**Review by gilbertbw — COMMENTED** [review]
Changes made in response to review

File: README.md [hunk 7]
```diff
@@ -41,20 +41,54 @@ Known issues are:
 - Navigation bars not supported
 - 64 bit Access has never been tested at all (doubt I really want to!)
 
-### Requirements
+# Requirements
```
[the remark this answers is not in the document]
Comment (bkp7) [reply README.md:40]: A bit more info.....

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (bkp7) [reply README.md:26]: A bit more info on later version support plus Ribbons etc. are supported...

File: Modules/modSCC.ACM [hunk 8]
```diff
@@ -3,9 +3,12 @@ Option Explicit
 
 Public Enum SccSystem
   sccSystemSubversion = 1  ' Supported
+  sccSystemText = 4        ' Supported
+  
+  ' These are placeholders for future possible enhanced support
+  ' for specific SCC systems
```
Comment (bkp7) [inline Modules/modSCC.ACM:9]:
```suggestion
  ' for specific SCC systems. The Text implementation does work generically
  ' with GIT but does not integrate fully, ie. specific git functions (commit, push, etc)
  ' have to be performed by the user
```

Comment (gilbertbw) [reply Modules/modSCC.ACM:13]: I have added a comment based off of this suggestion

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (gilbertbw) [reply README.md:26]: I have updated this in cc701f6906f494f80c43259b514eaa3680b3a180

**Comment by gilbertbw** [comment]
@bkp7 I think the only outstanding things on this are:

- Decide on a logo
- Decede what to do about the Company and Comments fields

**Comment by gilbertbw** [comment]
> 
> 
> @bkp7 I think the only outstanding things on this are:
> 
>     * Decide on a logo
> 
>     * Decede what to do about the Company and Comments fields

I have:
- moved setting Company and Comments fields out into #62 
- added my suggested image. For comparison here is:

1. master
2. as it was before readding the icon
3. now

image

here is the addin with these changes:
[AccessSCC.0.69.zip](https://github.com/x-ware-ltd/access-scc-addin/files/6115518/AccessSCC.0.69.zip)

**Comment by SWMahy** [comment]
Logo as it is now (3) looks ok to me.

**Review by SWMahy — APPROVED** [review]
Have not been able to perform a proper code review, but I understand I have been using this version of access-scc-addin for development work - with no issues, so have approved on that basis. I think the proposed logo look ok too.

**Review by bkp7 — APPROVED** [review]

**Pull request merged**